calblueprint/girls-write-now
# [auth] Implement login via username

[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

# What's new in this PR

- Changes the signup screen to include username, first name, last name, email, and password with input validation for username and email.
- Uses a database trigger to add user metadata to the profiles table
- Adds some basic styling to the sign up and log in page

[//]: # "Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

## Relevant Links

https://supabase.com/docs/guides/database/postgres/triggers
https://github.com/orgs/supabase/discussions/8383

### Notion Sprint Task

[Sprint task](https://www.notion.so/calblueprint/auth-Implement-login-via-username-5e3aaa0c6c0c42e3b31f162553865edf?pvs=4)

[//]: # 'Add the relevant Notion link(s) here'

### Online sources

[//]: # 'Optional - copy links to any tutorial or documentation that was useful to you when working on this PR'

### Related PRs

[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

## How to review

- Go to the sign up page
    - Try entering an invalid username and an already taken username to make sure both throw an error.
    - Enter an invalid email to make sure an error is shown on the screen
    - Create a new account and continue with the onboarding process
- Log out, and try logging in again with your new account
- Log out, try entering a username that does not exist
- Try entering a username that does exist with a wrong password

[//]: # 'The order in which to review files and what to expect when testing locally'

## Next steps
None

## Tests Performed, Edge Cases

See "How to Review"

### Screenshots

![IMG_3088](https://github.com/calblueprint/girls-write-now/assets/34043950/7e16b509-bead-4e53-ae03-81ad1778e72a)

@akshaynthakur
